### PR TITLE
New Cmus Cheat Sheet

### PR DESCRIPTION
**Pull request by rahul-raturi**
**What does your Instant Answer do?**
It provides a cheat sheet for cmus music player.

**What problem does your Instant Answer solve (Why is it better than organic links)?**
As cmus is a console based player, there are lots of keybindings. It's easier to remember when having a cheat sheet.

**What is the data source for your Instant Answer? (Provide a link if possible)**
https://github.com/cmus/cmus/blob/master/Doc/cmus.txt

**Why did you choose this data source?**
It is the official manual provided by cmus developers.

**Are there any other alternative (better) data sources?**
I don't know of.

**What are some example queries that trigger this Instant Answer?**
cmus cheat sheet, cmus guide

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Programmers, Music lovers, those who don't use GUI much, those who are fed up of their current music player.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
Yes.

**Which existing Instant Answers will this one supercede/overlap with?**
https://duck.co/ideas/idea/6369/cheat-sheet-for-cmus-c-music-player

**Are you having any problems? Do you need our help with anything?**
Any suggestions would be helpful.

**Where did you hear about DuckDuckHack? (For first time contributors)**
I saw it on my firefox browser search bar.

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
screenshot from 2015-09-12 08-54-13

## Checklist
Please place an 'X' where appropriate.

```
- [X] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [X] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [X] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 10

    - Windows 7
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 8
        - [] IE 9
        - [] IE 10

    - Windows XP
        - [] IE 7
        - [] IE 8

    - Mac OSX
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] Safari

    - iOS (iPhone)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - iOS (iPad)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - Android
        - [] Firefox
        - [] Native Browser
        - [] Google Chrome
        - [] Opera

```

-----
IA Page: https://duck.co/ia/view/cmus_cheat_sheet

### CODE REVIEW

File: share/goodie/cheat_sheets/json/cmus.json [hunk 1]
```diff
@@ -0,0 +1,163 @@
+{
+    "id": "6369",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/cmus.json]: let's make this a little more memorable! `cmus_cheat_sheet` sounds good.

Comment (moollaza) [inline share/goodie/cheat_sheets/json/cmus.json]: I'll update the IA Page `id` accordingly...

File: share/goodie/cheat_sheets/json/cmus.json [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    "name": "Cmus",
+    "description": "Console music player",
+    "metadata": {
+        "sourceName": "Cmus manual",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/cmus.json]: Let's change this to GitHub.

File: share/goodie/cheat_sheets/json/cmus.json [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            },
+            {
+                "val": "Remove track",
+                "key": "D"
+            }
+        ],
+        "Browser movements": [
+            {
+                "val": "Activate selection",
+                "key": "<space>"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/cmus.json]: you can remove the `<>` around `space` and `backspace`

File: share/goodie/cheat_sheets/json/cmus.json [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                "val": "Toggle repeat",
+                "key": "r"
+            },
+            {
+                "val": "Toggle playing mode i.e Artist/Album/All",
+                "key": "m"
+            },
+            {
+                "val": "Activate selection",
+                "key": "<enter>"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/cmus.json]: same here, let's remove the `<>`

**Comment by moollaza** [comment]
@rahul-raturi thanks a lot for contributing, we really appreciate it! We've left you some feedback that needs to be implemented. Once things are updated please let us know by leaving a comment, so we can review this again. If you have any questions, feel free to ask them here!

**Comment by rahul-raturi** [comment]
**@moollaza**, I've made the changes mentioned above. I guess it is ready for merge.

File: share/goodie/cheat_sheets/json/cmus.json [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+                "val": "Prepend track to play queue",
+                "key": "E"
+            },
+            {
+                "val": "Add track to playlist",
+                "key": "y"
+            },
+            {
+                "val": "Remove track",
+                "key": "D"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/cmus.json]: Does the casing matter here? Does this actually mean `[Shift] [D]` or just `d`?

If casing doesn't matter, then let's capitalize all the single letters. If casing **does** matter, then let's add `[Shift] [D]` as appropriate.

File: share/goodie/cheat_sheets/json/cmus.json [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+            }
+        ],
+        "Browser movements": [
+            {
+                "val": "Activate selection",
+                "key": "space"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/cmus.json]: Let's capitalize Space, Backspace and Enter

**Comment by moollaza** [comment]
@rahul-raturi just a couple more things and this is done. Thanks!

File: share/goodie/cheat_sheets/json/cmus.json (hunk 5 above)
Comment (rahul-raturi) [inline share/goodie/cheat_sheets/json/cmus.json]: Casing does matters in cmus. I'm updating the cheat sheet to include suggested changes.

**Comment by rahul-raturi** [comment]
**@moollaza**, I've included suggested changes and some new ones too (which I forgot in first place). 
screenshot from 2015-09-17 19-26-17-cropped
In the screenshot above, when there is a simultaneous key with **shift** (like **shift e**), they are separated by a line break. Is there a way to make them inline.? Or, should it be left as it is?

**Comment by moollaza** [comment]
@rahul-raturi this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

**Comment by moollaza** [comment]
>In the screenshot above, when there is a simultaneous key with shift (like shift e), they are separated by a line break. Is there a way to make them inline.? Or, should it be left as it is?

@rahul-raturi you can leave as-is for now. We're working on making these look a little better so that things like this don't happen for all cheat sheets.

**Comment by moollaza** [comment]
@rahul-raturi congrats! It's live: https://duckduckgo.com/?q=cmus+cheat+sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Pull request merged**